sky/serve/core.py

```diff
@@ -258,10 +293,15 @@ def up(
         else:
             lb_port = serve_utils.load_service_initialization_result(
                 lb_port_payload)
-            endpoint = backend_utils.get_endpoints(
+            socket_endpoint = backend_utils.get_endpoints(
                 controller_handle.cluster_name, lb_port,
                 skip_status_check=True).get(lb_port)
-            assert endpoint is not None, 'Did not get endpoint for controller.'
+            assert socket_endpoint is not None, (
+                'Did not get endpoint for controller.')
+            # Already checked by _validate_service_task
+            assert task.service is not None
+            schema = 'http' if task.service.tls_credential is None else 'https'
```

Comment:
```suggestion
            protocol = 'http' if task.service.tls_credential is None else 'https'
```
It seems HTTP / HTTPS are more commonly called `protocol`?